### ISSUE TITLE
Dependency Error: datatables.net-vue3@2.1.3 is not compatible with datatables.net@2.0.0

### ISSUE THREAD

**Issue by mrorabau** [DataTables/Vue#29]
datatables.net-vue3@2.1.3 (^2.1.3) is not compatible with datatables.net@2.0.0 (^2.0.0). Adding --legacy-peer-deps to npm seems to work, so it could perhaps just be a dependency limitation in the vue3 component.

**Comment by mrorabau** [comment]
From: https://github.com/DataTables/Vue/blob/main/package.json

![image](https://github.com/DataTables/Vue/assets/1796227/92b02fc6-e4af-43fa-8879-db8d0ec40d1f)

**Comment by mrorabau** [comment]
See PR https://github.com/DataTables/Vue/pull/30

**Comment by AllanJard** [comment]
Thanks - I'm not going to pull it in just yet, and I've made changes to the Vue component for DataTables which includes support for components in the table cells, but I'd like to make it more flexible than what it currently is. [I've got this thread open with the Vue core](https://github.com/vuejs/core/discussions/10282) to see if there are any options. With that resolved I'll be looking to get this release done.

**Comment by gy-jamesb** [comment]
I wanted to add a note that the PR #30 will fix but it was causing me frustrating issues.  When I added the extension datatables.net-buttons-bs5 to my project I would get an error stating that Datatables.Buttons.default was undefined. In addition to this using the datatables.net-buttons extension worked but broke all the styles, including pagination.  When I checked the version of against datatables.net-bs5 and datatables.net I noticed that npm install --save datatables.net-vue3  was pulling in version 1.13.11 for datatables.net and that was throwing off everything.  I am sure if I added all these references manually like I have now this would not have been an issue, but the docs site just states run the one install.  Again, this will be fixed once the PR is completed but I wanted to let others know if they have a ghost issue trying to use extensions.

**Comment by BayBreezy** [comment]
Hey @AllanJard , Is the component feature working as yet? If so, is there an example that I can follow?

Trying to use the vue docs but it's not rendering the button for me

**Comment by AllanJard** [comment]
Yes - use the 3.x release of this package. Documentation for how to work with Vue components is [available here](https://datatables.net/manual/vue#Vue-Components).

**Comment by AllanJard** [comment]
Can you create a test case showing the issue please?

> P.S: the responsive & select are not in the Cinfig interface that is exported from datatables.net

No they aren't. They are added by `datatables.net-responsive` and `datatables.net-select` respectively.

**Comment by BayBreezy** [comment]
@AllanJard Here is the link: https://stackblitz.com/edit/nuxt-starter-7llxe7?file=pages%2Fnope.vue

Stackblitz is not rendering the table properly because of some import issue with the core DT package.

If you loop at the nope.vue file, you see where the columns are passed in via the options object. When done like this, the component in the action slot is not rendered.

**Comment by AllanJard** [comment]
I see what you mean. Thanks for the clarification.

[Fix committed here](https://github.com/DataTables/Vue/commit/06d92e0b2d30817261fd7ac966cafc754cd16d7f).

**Comment by BayBreezy** [comment]
TY!

Btw, this feature is only compatible with v2.x of datatables, correct?

**Comment by AllanJard** [comment]
Yup, v2 only.

**Comment by BayBreezy** [comment]
@AllanJard Cool cool.

Looking forward to the release. 
I managed to get it working in my UI Thing: https://ui-thing.behonbaker.com/components/datatable#custom-component

I have not studied the whole new `layout` thing just yet.. But I will get there
